easybuild/easyblocks/generic/juliapackage.py

```diff
@@ -0,0 +1,170 @@
+                env.setvar('JULIA_PKG_OFFLINE', 'true')
+            else:
+                errmsg = (
+                    "Cannot set offline mode in Julia v%s (needs Julia >= 1.5). "
+                    "Enable easyconfig option 'bundle_deps' to allow installation "
+                    "with any extra downloaded dependencies."
+                )
+                raise EasyBuildError(errmsg, julia_version)
+
+    def prepare_step(self, *args, **kwargs):
+        """Prepare for installing Julia package."""
+        super(JuliaPackage, self).prepare_step(*args, **kwargs)
+        self.set_pkg_offline()
+
+    def configure_step(self):
+        """No separate configuration for JuliaPackage."""
+        pass
+
+    def build_step(self):
+        """No separate build procedure for JuliaPackage."""
+        pass
+
+    def test_step(self):
+        """No separate (standard) test procedure for JuliaPackage."""
+        pass
+
+    def install_julia_pkg(self):
+        """Install procedure for Julia packages with Pkg"""
+
+        # prepend installation directory to Julia DEPOT_PATH
+        # see https://docs.julialang.org/en/v1/manual/environment-variables/#JULIA_DEPOT_PATH
+        env.setvar('JULIA_DEPOT_PATH', ':'.join([self.installdir, os.getenv('JULIA_DEPOT_PATH', '')]))
+
+        # command sequence for Julia.Pkg
+        julia_pkg_cmd = ['using Pkg']
+        if os.path.isdir(os.path.join(self.start_dir, '.git')):
+            # sources from git repos can be installed as any remote package
+            self.log.debug('Installing Julia package in normal mode (Pkg.add)')
+
+            julia_pkg_cmd.extend([
+                # install package from local path preserving existing dependencies
+                'Pkg.add(url="%s"; preserve=Pkg.PRESERVE_ALL)' % self.start_dir,
+            ])
+        else:
+            # plain sources have to be installed in develop mode
+            # copy sources to install directory and install
+            self.log.debug('Installing Julia package in develop mode (Pkg.develop)')
+
+            install_pkg_path = os.path.join(self.installdir, 'packages', self.name)
+            copy_dir(self.start_dir, install_pkg_path)
+
+            julia_pkg_cmd.extend([
+                'Pkg.develop(PackageSpec(path="%s"))' % install_pkg_path,
+                'Pkg.build("%s")' % self.name,
+            ])
+        julia_pkg_cmd = ';'.join(julia_pkg_cmd)
+
+        install_cmd = "julia -e '%s'" % julia_pkg_cmd
+
+        cmd = ' '.join([self.cfg['preinstallopts'], install_cmd, self.cfg['installopts']])
+
+        (out, _) = run_cmd(cmd, log_all=True, simple=False)
+
+        return out
+
+    def install_step(self):
+        """Install Julia package"""
+        self.install_julia_pkg()
+
+    def run(self):
+        """Install Julia package as an extension."""
+
+        if not self.src:
+            errmsg = "No source found for Julia package %s, require    d for installation. (src: %s)"
```

Comment:
`require    d` => `required`

Reply:
Fixed in https://github.com/easybuilders/easybuild-easyblocks/pull/2816/commits/a8e23e5b8a9dac5de6a603e829229e54841e1c8c